### PR TITLE
Ensure source bank is resampled at each generation for fixed source

### PR DESCRIPTION
**Pull request by paulromano**
As a [user noted](https://groups.google.com/forum/#!topic/openmc-users/rVdmODgNuNk) on the user's mailing list, the source bank is (incorrectly) the same for every generation in a fixed source simulation. This pull request fixes that, as well as two other things:

1. The documentation for `<source_point write=... />` incorrected said "source_write".
2. Reading particle restart files incorrectly expected a "energy_group" dataset.

### CODE REVIEW

**Comment by paulromano** [comment]
I also added a commit which fixes [a bug with Ni-59](https://groups.google.com/forum/#!topic/openmc-users/VFtJ_gvWYLM). The problem is that Ni-59 is the only nuclide which has a MT=3 cross section in the ACE file. OpenMC doesn't recognize this and ends up adding it to the total cross section even though it is redundant. The result is an eventual segfault.

File: src/constants.F90 [hunk 1]
```diff
@@ -207,7 +207,7 @@ module constants
        N_4N2P  = 194, N_4N2A = 195, N_4NPA  = 196, N_3P    = 197, N_N3P   = 198, &
        N_3N2PA = 199, N_5N2P = 200, N_P0    = 600, N_PC    = 649, N_D0    = 650, &
        N_DC    = 699, N_T0   = 700, N_TC    = 749, N_3HE0  = 750, N_3HEC  = 799, &
-       N_A0    = 800, N_AC   = 849, N_2N0   = 875, N_2NC   = 891
+       N_A0    = 800, N_AC   = 849, N_2N0   = 875, N_2NC   = 891, N_NONELASTIC = 3
```
Comment (smharper) [inline src/constants.F90]: It'd be better to insert this after elastic so the order is preserved.

**Comment by smharper** [comment]
Looks good. Merged!

**Comment by paulromano** [comment]
Appreciate the quick review!

**Pull request merged**